test/extended/security/labels.go

```diff
@@ -0,0 +1,62 @@
+package security
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func matchForbiddenNodeLine(nodeName, labelName, outPut string) bool {
+	regexTemplate := fmt.Sprintf(`Error from server \(Forbidden\): nodes "%s" is forbidden: is not allowed to modify labels: %s`,
+		regexp.QuoteMeta(nodeName), regexp.QuoteMeta(strings.Split(labelName, "=")[0]))
+	pattern := regexp.MustCompile(regexTemplate)
+	return pattern.MatchString(outPut)
+}
+
+// testOpenshiftNodeLabeling attempts to apply the label to the given node by using node's
+// kubeconfig present in /var/lib/kubelet/kubeconfig. It will capture the stdout/stderr of that execution
+// and process it to verify that the forbidden labels aren't getting applied.
+func testOpenshiftNodeLabeling(oc *exutil.CLI, node *corev1.Node, forbiddenLabel string) bool {
+	// `oc debug` command below attempts to set a forbidden label. This command is expected to fail. However,
+	// during testing it is observed that sometimes even though the command fails to set the forbidden label on the node
+	// it does not return a valid error and in fact returns with exit code 0. This ends up making
+	// the test flaky. Because of this observed flakiness, we are ignoring the error returned by the
+	// `oc debug` command below, and rely on the regex matching in matchForbiddenNodeLine function
+	// to decide if the node was successfully prevented from labeling the forbidden label.
+	out, _ := oc.AsAdmin().Run("debug").Args("node/"+node.Name, "--", "chroot", "/host", "/bin/bash", "-c",
+		fmt.Sprintf(`KUBECONFIG=/var/lib/kubelet/kubeconfig kubectl label nodes/%s %s `, node.Name, forbiddenLabel)).Output()
+
+	return matchForbiddenNodeLine(node.Name, forbiddenLabel, out)
+}
+
+var _ = g.Describe("[sig-node] [Conformance] Prevent openshift node labeling on update by the node", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithPodSecurityLevel("node-label-e2e", admissionapi.LevelPrivileged)
+
+	g.It("TestOpenshiftNodeLabeling", func() {
+		clusterAdminKubeClientset := oc.AdminKubeClient()
+
+		workerNodes, err := clusterAdminKubeClientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{
```

Comment:
might need to use wait.ExponentialBackoff here too to make sure we get the Items populated.